### PR TITLE
Added HMAC-SHA256 authentication for uplink commands

### PR DESCRIPTION
**Pull request by saksham2001**
According to IARU requirement the uplink commands need authentication. I have added SHA256 based authentication where the Argus and the ground station will have a shared private key and all the uplink commands will have a MAC generated using the command and nonce (random number). The satellite will generate this MAC again using the received command and nonce, and make sure it matches the received MAC.

### Changed files
- flight/apps/comms/auth.py (new) — HMAC-SHA256 computation, constant-time MAC comparison, packet verification
- flight/core/hashlib.py (new) — Pure-Python SHA256 for CircuitPython. Ported from [Adafruit's Circuit Python Hashlib Library](https://github.com/adafruit/Adafruit_CircuitPython_hashlib)
- flight/apps/comms/comms.py — Integrates auth verification into receive_message(), extracts payload without nonce/MAC trailer
- flight/tasks/comms.py — Distinguishes auth failure from invalid command in logging
- flight/configuration/ground.yaml — Adds AUTH_ENABLED and AUTH_KEY_HEX
- flight/configuration/flight.yaml — Adds AUTH_ENABLED and AUTH_KEY_HEX
- tests/test_command_auth.py (new) — Unit tests validating HMAC against Python stdlib, success/failure/length/key edge cases

### Testing
- Tested on flatsat and the other build in lab, with correct and incorrect key.

### Summary
- Adds command authentication using HMAC-SHA256 with a 256-bit shared key to verify all uplink commands from the ground station
- Authenticated packets include a 4-byte random nonce and 32-byte MAC trailer appended after command args
- Falls back to legacy unauthenticated parsing when AUTH_ENABLED is false, preserving backwards compatibility
- Adds pure-Python SHA256 hashlib (Adafruit CircuitPython port) for MicroPython compatibility
- Auth config added to both ground.yaml and flight.yaml

### Notes
We will have to change the private keys on ground station and flight builds before flight.

### CODE REVIEW

File: flight/configuration/ground.yaml [hunk 1]
```diff
@@ -41,4 +41,8 @@ comms:
   ARGUS_ID:
     value: 0
     _const: true
-    _hex: true
\ No newline at end of file
+    _hex: true
+  AUTH_ENABLED:
+    value: true
+  AUTH_KEY_HEX:
+    value: "d6172b38acb7d2a28e21662f689d1d15ad78ccc888a9c7a78ef58cb61b0f1e32" # Change before flight
```
Comment (Mahii11) [inline flight/configuration/ground.yaml]: Add TODO flag for this too

**Review by sophizhao — APPROVED** [review]

**Review by sophizhao — APPROVED** [review]
